test/extended/rorfs/readonly_root_filesystem.go

```diff
@@ -0,0 +1,380 @@
+			_, err := oc.AdminKubeClient().CoreV1().Namespaces().Get(context.Background(), ns, metav1.GetOptions{})
+			if err == nil {
+				namespaces = append(namespaces, ns)
+				e2e.Logf("Namespace %s exists, will be tested", ns)
+			} else {
+				e2e.Logf("Namespace %s does not exist, skipping", ns)
+			}
+		}
+
+		if len(namespaces) == 0 {
+			g.Skip("No testable namespaces found in this cluster")
+		}
+
+		oc.SetupProject()
+
+		for _, ns := range namespaces {
+			// Get pods using optimized function with label selector to exclude unwanted pods
+			labelSelector := "app notin (guard,installer,pruner,revision)"
+			podNames, err := exutil.GetPodsListByLabel(oc.AsAdmin(), ns, labelSelector)
+			if err != nil {
+				e2e.Failf("Failed to get pods from namespace %s: %v", ns, err)
+			}
+
+			if len(podNames) == 0 {
+				e2e.Logf("No suitable pods found in namespace %s, skipping", ns)
+				continue
+			}
+
+			// Filter out pods that should be skipped based on name patterns
+			var filteredPodNames []string
+			skipPatterns := []string{
+				"apiserver-watcher",
+			}
+
+			for _, podName := range podNames {
+				// Skip pods with specific name patterns
+				shouldSkip := false
+				for _, pattern := range skipPatterns {
+					if strings.Contains(strings.ToLower(podName), strings.ToLower(pattern)) {
+						e2e.Logf("Skipping pod %s in namespace %s (matches exclusion pattern: %s)", podName, ns, pattern)
+						shouldSkip = true
+						break
+					}
+				}
+				if shouldSkip {
+					continue
+				}
+				filteredPodNames = append(filteredPodNames, podName)
+			}
+
+			if len(filteredPodNames) == 0 {
+				e2e.Logf("No suitable pods found in namespace %s after filtering, skipping", ns)
+				continue
+			}
+
+			e2e.Logf("Found %d pods to test in namespace %s (after filtering)", len(filteredPodNames), ns)
+			for _, podName := range filteredPodNames {
+				err := exutil.AssertPodToBeReady(oc, podName, ns)
+				if err != nil {
+					e2e.Logf("Pod %s in namespace %s is not ready: %v, skipping", podName, ns, err)
+					continue
+				}
+
+				e2e.Logf("Inspect the %s Pod's securityContext.", ns)
+				pod, err := oc.AdminKubeClient().CoreV1().Pods(ns).Get(context.Background(), podName, metav1.GetOptions{})
+				if err != nil {
+					e2e.Failf("Failed to get pod %s details: %v", podName, err)
+				}
+				if pod.Spec.Containers[0].SecurityContext == nil {
+					e2e.Failf("Pod %s in namespace %s does not have securityContext", podName, ns)
+				}
+				if pod.Spec.Containers[0].SecurityContext.ReadOnlyRootFilesystem == nil {
+					e2e.Failf("Pod %s in namespace %s does not have readOnlyRootFilesystem set", podName, ns)
+				}
+
+				// Check readOnlyRootFilesystem value
+				isReadOnly := *pod.Spec.Containers[0].SecurityContext.ReadOnlyRootFilesystem
+				if isReadOnly {
+					e2e.Logf("Pod %s in namespace %s has readOnlyRootFilesystem=true", podName, ns)
+				} else {
+					e2e.Logf("Pod %s in namespace %s has readOnlyRootFilesystem=false", podName, ns)
+				}
```

Comment:
Since `readOnlyRootFilesystem` is at the level of the individual container's security context and not the pod-level security context, shouldn't we be checking each container for it instead of only the first one? That way, we'd catch sidecars as well.